core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -21,27 +21,37 @@ oppia.directive('collectionEditorNavbar', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_editor_navbar_directive',
     controller: [
-      '$scope', '$modal', 'alertsService', 'routerService', 'UndoRedoService',
-      'CollectionEditorStateService', 'CollectionValidationService',
-      'CollectionRightsBackendApiService',
-      'EditableCollectionBackendApiService',
+      '$scope', '$log', '$modal', '$rootScope', 'alertsService',
+      'routerService', 'UndoRedoService', 'CollectionEditorStateService',
+      'CollectionRightsBackendApiService', 'CollectionRightsObjectFactory',
+      'CollectionValidationService', 'EditableCollectionBackendApiService',
       'EVENT_COLLECTION_INITIALIZED', 'EVENT_COLLECTION_REINITIALIZED',
       'EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED',
       function(
-          $scope, $modal, alertsService, routerService, UndoRedoService,
-          CollectionEditorStateService, CollectionValidationService,
-          CollectionRightsBackendApiService,
-          EditableCollectionBackendApiService,
+          $scope, $log, $modal, $rootScope, alertsService,
+          routerService, UndoRedoService, CollectionEditorStateService,
+          CollectionRightsBackendApiService, CollectionRightsObjectFactory,
+          CollectionValidationService, EditableCollectionBackendApiService,
           EVENT_COLLECTION_INITIALIZED, EVENT_COLLECTION_REINITIALIZED,
           EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED) {
+        var collectionRightsObject = null;
+        $scope.rightsHasLoaded = false;
+
         $scope.collectionId = GLOBALS.collectionId;
         $scope.collection = CollectionEditorStateService.getCollection();
-        $scope.isPrivate = GLOBALS.isPrivate;
-        $scope.canUnpublish = GLOBALS.canUnpublish;
         $scope.validationIssues = [];
         $scope.isSaveInProgress = (
           CollectionEditorStateService.isSavingCollection);
 
+        CollectionRightsBackendApiService.getCollectionRights(
+          $scope.collectionId).then(function(data) {
+          $scope.isPrivate = data.is_private;
```

Comment:
Do you need to maintain $scope.isPrivate etc. when you can refer to collectionRights.isPrivate (and yep, I think it's ok to call it collectionRights rather than collectionRightsObject)?

I'm fine either way, just wondered if it was worth saving the extra variable (since then you have yet another thing to keep in sync).